shell/src/main/java/tachyon/shell/command/CommandUtils.java

```diff
@@ -0,0 +1,157 @@
+import java.util.List;
+
+import tachyon.Constants;
+import tachyon.TachyonURI;
+import tachyon.client.file.TachyonFile;
+import tachyon.client.file.TachyonFileSystem;
+import tachyon.client.file.options.SetStateOptions;
+import tachyon.client.lineage.TachyonLineage;
+import tachyon.exception.TachyonException;
+import tachyon.thrift.FileInfo;
+import tachyon.thrift.LineageInfo;
+import tachyon.util.FormatUtils;
+
+/**
+ * Common util methods for executing commands.
+ */
+public class CommandUtils {
+
+  private CommandUtils() {
+    // Not intended for instantiation.
+  }
+
+  /**
+   * Set a new TTL value or unset an existing TTL value for file at path.
+   *
+   * @param path the file path
+   * @param ttlMs the TTL (time to live) value to use; it identifies duration (in milliseconds) the
+   *        created file should be kept around before it is automatically deleted, irrespective of
+   *        whether the file is pinned; {@link Constants#NO_TTL} means to unset the TTL value
+   * @throws IOException when failing to set/unset the TTL
+   */
+  public static void setTTL(TachyonFileSystem tfs, TachyonURI path, long ttlMs) throws IOException {
+    try {
+      TachyonFile fd = tfs.open(path);
+      SetStateOptions options = new SetStateOptions.Builder().setTTL(ttlMs).build();
+      tfs.setState(fd, options);
+    } catch (TachyonException e) {
+      throw new IOException(e.getMessage());
+    }
+  }
+
+  /**
+   * Displays information for all directories and files directly under the path specified in argv.
+   *
+   * @param tfs The TachyonFileSystem client
+   * @param path The TachyonURI path as the input of the command
+   * @param recursive Whether list the path recursively
+   * @throws IOException
+   */
+  public static void ls(TachyonFileSystem tfs, TachyonURI path, boolean recursive)
+      throws IOException {
+    List<FileInfo> files = listStatusSortedByIncreasingCreationTime(tfs, path);
+    String format = "%-10s%-25s%-15s%-15s%-5s%n";
+    for (FileInfo file : files) {
+      String inMemory = "";
+      if (!file.isFolder) {
+        if (100 == file.inMemoryPercentage) {
+          inMemory = "In Memory";
+        } else {
+          inMemory = "Not In Memory";
+        }
+      }
+      System.out.format(format, FormatUtils.getSizeFromBytes(file.getLength()),
+          convertMsToDate(file.getCreationTimeMs()), inMemory, file.getUserName(),
+          file.getPath());
+      if (recursive && file.isFolder) {
+        ls(tfs, new TachyonURI(path.getScheme(), path.getAuthority(), file.getPath()), true);
+      }
+    }
+  }
+
+  /**
+   * Convert a millisecond number to a formatted date String
+   * @param millis a long millisecond number
+   * @return formatted date String
+   */
+  public static String convertMsToDate(long millis) {
+    DateFormat formatter = new SimpleDateFormat("MM-dd-yyyy HH:mm:ss:SSS");
+    return formatter.format(new Date(millis));
+  }
+
+  public static void listLineages() throws IOException {
```

Comment:
Yep, I understand it. Is it possible to add abstract class to group commands and avoid commonUtils? For example, the `listLineage` might be put into a `LineageCommand` super class.